### PR TITLE
Frontend MVP

### LINKED ISSUE #1: Add map
We need a map so that we can show where the aircraft is and where targets, obstacles, and other teams' aircraft are.

At the very minimum, the map should use satellite view; I highly recommend Mapbox.

### LINKED ISSUE #2: Show aircraft on map
As a pilot, I'd like to see where the aircraft is on the map.

This is a good story to get started with drawing points on the map, communicating with the telemetry service, and debugging the communication.

### LINKED ISSUE #3: Show simulated PFD for aircraft
As a pilot, I'd like to see a simulated version of a typical primary flight display (PFD) for the aircraft. A PFD typically includes attitude, airspeed, altitude, and heading, but we can also stick in battery and latency information.

### LINKED ISSUE #4: Show warning panel
A warning panel is necessary to indicate abnormal conditions in the infrastructure, aircraft, or communication between frontend and infrastructure. The rationale is that we don't expect the pilot to look at console logs for errors, or to have to "realize" that something is wrong - if there is an error or warning presently affecting the aircraft (or the frontend's presentation of the aircraft), it should be indicated.

JavaScript errors are a little special, but we don't expect many to pop up as much as we used to have compared to the previous frontend (since we're using TypeScript, which should eliminate many type errors). Frontend code errors will likely fall into arithmetic or race conditions.

A piece of the UI should be cut to provide space for this panel.

### PR DESCRIPTION
**Pull request by sshh12**
#### This PR
Map, plane(s), service integrations, warnings/logs, gauges
* Fixes #4 
* Fixes #3
* Fixes #2
* Fixes #1
#### Future PRs
* Portable version
* Draw more stuff on the map
* Offline support
* "Path Find" integration
* More restrictive typing (less `any` type)

### CODE REVIEW

File: src/components/Map.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  render() {
+    let { width, height } = this.state;
+    return (
+      <div className="map-wrapper">
+        <ServicesContext.Consumer>
+          {({ telemetry, mission, odlcs }) => {
+            if (telemetry) {
+              // @ts-ignore
```
Comment (oldmud0) [inline src/components/Map.tsx]: What happened here

File: src/components/SideView.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    <Button.Or />
+    <Button positive><Icon name='sync' />Refresh&nbsp;&nbsp;&nbsp;&nbsp;</Button>
+  </Button.Group>;
+};
+
+const LOG_TYPE_TO_COLOR: any = {
+  warn: '#fbbd08',
+  error: '#db2828',
+  info: '#64615f'
+};
```
Comment (oldmud0) [inline src/components/SideView.tsx]: Feel like this should trigger a type error if a new log level is added somewhere else.

File: src/index.tsx [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 import './index.css';
 import App from './App';
 import * as serviceWorker from './serviceWorker';
 
+declare const FlightIndicator: any;
+FlightIndicator.setOptions({
+  assets: "/pfd/"
+});
+
+mapboxgl.accessToken = 'pk.eyJ1Ijoic3NoaDEyIiwiYSI6ImNpcTVhNDQxYjAwM3FmaGtrYnl6czEwMGcifQ.eYETiDD8NqThLahLIBmjSQ';
```
Comment (oldmud0) [inline src/index.tsx]: todo: make this configurable

File: src/components/Map.tsx (hunk 1 above)
Comment (sshh12) [reply src/components/Map.tsx]: The ts-ignore? The mapbox types are a lil broken so it doesn't like the next line.

**Comment by SteveHan-233** [comment]
I'm currently kind of stuck on the offline thing - the front end is not even sending any requests... But I also realized that the Chrome cache handles the offline map really well. I'm pretty sure we can just use that.

**Comment by oldmud0** [comment]
Unfortunately, the browser cache cannot guarantee that all our tiles are stored offline when we need them. It's also not trivial copying the cache to another machine.

**Pull request merged**